File: app/src/main/java/com/example/babajidemustapha/survey/shared/room/StringListConverter.java
package com.example.babajidemustapha.survey.shared.room;

import androidx.room.TypeConverter;

import com.google.gson.Gson;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class StringListConverter {
    @TypeConverter
    public static List<String> fromString(String value) {
        if (value != null) {
            return new ArrayList<>(Arrays.asList(new Gson().fromJson(value, String[].class)));
        }
        return new ArrayList<>();
    }

    @TypeConverter
    public static String fromArrayList(List<String> list) {
        return list == null || list.isEmpty() ? null : new Gson().toJson(list);
    }
}

File: app/src/main/java/com/example/babajidemustapha/survey/shared/room/StringListConverter.kt
package com.example.babajidemustapha.survey.shared.room

import androidx.room.TypeConverter
import com.google.gson.Gson
import java.util.*

object StringListConverter {
    @kotlin.jvm.JvmStatic
    @TypeConverter
    fun fromString(value: String?): List<String> {
        return if (value != null) {
            ArrayList(Arrays.asList(*Gson().fromJson(value, Array<String>::class.java)))
        } else ArrayList()
    }

    @kotlin.jvm.JvmStatic
    @TypeConverter
    fun fromArrayList(list: List<String?>?): String? {
        return if (list == null || list.isEmpty()) null else Gson().toJson(list)
    }
}